-- a/webapp/graphite/render/functions.py
+++ b/webapp/graphite/render/functions.py
@@ -594,7 +594,8 @@
   if type(nodes) is int:
     nodes=[nodes]
   for series in seriesList:
-    newname = re.search('(?:.*\()?(?P<name>[\w\.]+)(?:,|\]?.*)?',series.name).groups()[0]
+    name_without_parens = re.search('(?:.*\()?(?P<name>[\w\.]+)(?:,|\]?.*)?',series.name).groups()[0]
+    newname = ".".join([ name_without_parens.split(".")[node] for node in nodes])
     series.name = newname
   return seriesList

